STYLEGUIDE.adoc

```diff
@@ -0,0 +1,541 @@
+= Antaeus Haskell style guide
+:toc:
+
+The purpose of this document is to codify various aspects of how we write Haskell code as we contribute to Antaeus.
+
+The goal is twofold:
+
+1. To provide guidance in cases where it is not obvious what to do, or the choice is arbitrary.
+2. To help resolve disagreements between team members. Ideally these can be settled by referring to a recommendation in this document, or if not, then perhaps the resolution of the disagreement will create a new recommendation.
+
+The contents of this document are “recommendations”, that is:
+
+- They are more concrete than “principles” (like “write simple code”) because we want them to be clearly action-guiding.
+
+- They are more vague than “rules” (like “never use function X”) because it’s usually impossible to be that precise, and so judgement is still required in applying them.
+
+== Haskell
+
+=== Basic formatting
+
+Use 2 spaces for indentation. Full-indent for `where` is recommended.
+
+Bad:
+
+[source,Haskell]
+----
+f x = y
+ where
+  y = ...
+----
+
+Good:
+
+[source,Haskell]
+----
+f x = y
+  where
+    y = ...
+----
+
+Avoid lines over 100 characters. This is codified in `.editorconfig`.
```

Comment:
I'm currently using 120 chars, which is fine. Sometimes hard coded bytestrings are long and need to concat them, which is a bit tedious.

I'm happy to move to 100 chars if it's what the team prefers.